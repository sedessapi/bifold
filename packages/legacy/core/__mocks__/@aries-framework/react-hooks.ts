/* eslint-disable @typescript-eslint/no-explicit-any */
import { LegacyIndyCredentialFormat } from '@aries-framework/anoncreds'
import {
  BasicMessageRecord,
  CredentialExchangeRecord,
  CredentialProtocolOptions,
  ProofExchangeRecord,
} from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ records: [] } as any)
const useProofs = jest.fn().mockReturnValue({ records: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialExchangeRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofExchangeRecord[])
const useBasicMessagesByConnectionId = jest.fn().mockReturnValue([] as BasicMessageRecord[])
const useBasicMessages = jest.fn().mockReturnValue({ records: [] as BasicMessageRecord[] })
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
  getFormatData: jest
    .fn()
    .mockReturnValue(
      Promise.resolve({} as CredentialProtocolOptions.GetCredentialFormatDataReturn<[LegacyIndyCredentialFormat]>)
    ),
}
const mockProofModule = {
  getCredentialsForRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
  getFormatData: jest.fn(),
  findRequestMessage: jest.fn(),
  requestProof: jest.fn(),
  update: jest.fn(),
}

const mockMediationRecipient = {
  initiateMessagePickup: jest.fn(),
}
const mockOobModule = {
  findById: jest.fn().mockReturnValue(Promise.resolve(null)),
  createInvitation: jest.fn(),
  toUrl: jest.fn(),
}
const mockBasicMessageRepository = {
  update: jest.fn(),
}
const mockAgentContext = {
  dependencyManager: {
    resolve: jest.fn().mockReturnValue(mockBasicMessageRepository),
  },
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
    mediationRecipient: mockMediationRecipient,
    oob: mockOobModule,
    context: mockAgentContext,
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()
const useConnections = jest.fn()

export {
  useAgent,
  useConnectionById,
  useCredentials,
  useProofs,
  useCredentialById,
  useCredentialByState,
  useProofById,
  useProofByState,
  useConnections,
  useBasicMessages,
  useBasicMessagesByConnectionId,
}
