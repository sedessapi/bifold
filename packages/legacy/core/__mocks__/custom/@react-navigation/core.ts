const navigate = jest.fn()
const dispatch = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  setOptions: jest.fn(),
  getParent: () => {return {
    navigate,
    dispatch,
  }},
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
