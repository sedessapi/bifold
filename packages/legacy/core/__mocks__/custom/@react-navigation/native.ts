// import { useFocusEffect } from '@react-navigation/native'

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

const CommonActions = {
  reset: jest.fn(),
}

const useIsFocused = () => {
  return true
}

export { useIsFocused, useFocusEffect, createNavigatorFactory, CommonActions }
