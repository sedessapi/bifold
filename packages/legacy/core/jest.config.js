module.exports = {
  preset: 'react-native',
  testTimeout: 10000,
  setupFiles: ['<rootDir>/jestSetup.js'],
  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  moduleNameMapper: {
    '\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/__mocks__/file.js',
    '\\.(css|less)$': '<rootDir>/__mocks__/style.js',
    axios: require.resolve('axios'),
    'react-i18next': '<rootDir>/__mocks__/react-i18next.ts',
    '^uuid$': require.resolve('uuid'),
    '@aries-framework/core': require.resolve('@aries-framework/core'),
    '@aries-framework/anoncreds': require.resolve('@aries-framework/anoncreds'),
  },
  transform: {
    '^.+\\.(js|jsx|ts|tsx)$': 'babel-jest',
  },
  transformIgnorePatterns: [
    'node_modules\\/(?!(.*react-native.*)|(uuid)|(@aries-framework\\/core)|(@aries-framework\\/anoncreds))',
  ],
  testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$',
  testPathIgnorePatterns: [
    '\\.snap$',
    '<rootDir>/node_modules/',
    '<rootDir>/lib',
    '<rootDir>/__tests__/contexts/',
    '<rootDir>/__tests__/helpers/',
  ],
  coveragePathIgnorePatterns: [
    '/node_modules/',
    '/__tests__/',
    '<rootDir>/App/navigators/defaultStackOptions.tsx',
    '<rootDir>/App/defaultConfiguration.ts',
    '<rootDir>/App/components/buttons/InfoIcon.tsx',
    '<rootDir>/App/hooks/deep-links.ts',
  ],
  cacheDirectory: '.jest/cache',
  snapshotFormat: {
    escapeString: true,
    printBasicPrototype: true,
  },
}
