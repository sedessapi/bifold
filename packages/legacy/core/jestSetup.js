/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import 'reflect-metadata'
import 'react-native-gesture-handler/jestSetup'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock.js'
import mockRNLocalize from 'react-native-localize/mock'

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('react-native-localize', () => mockRNLocalize)
