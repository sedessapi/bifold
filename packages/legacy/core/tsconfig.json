{
  "extends": "../../../tsconfig-base.json",
  "compilerOptions": {
    "baseUrl": "./App",
    "outDir": "./lib"
  },
  "exclude": ["node_modules", "babel.config.js", "jest.config.js", "jestSetup.js", "lib"]
}