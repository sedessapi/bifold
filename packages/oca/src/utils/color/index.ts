import generateColor from './generateColor'
// Remove hashCode and hashToRGBA when deprecated OCA resolver is removed
import hashCode from './hashCode'
import hashToRGBA from './hashToRGBA'
import { textColorForBackground } from './textColorForBackground'

export { generateColor, textColorForBackground, hashCode, hashToRGBA }
