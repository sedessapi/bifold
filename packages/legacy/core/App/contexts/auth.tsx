// NOTE: We need to import these to be able to use the AskarWallet in this file.
import '@hyperledger/aries-askar-react-native'
// eslint-disable-next-line import/no-extraneous-dependencies
import 'reflect-metadata'

import { AskarWallet } from '@aries-framework/askar'
import { ConsoleLogger, LogLevel, SigningProviderRegistry } from '@aries-framework/core'
import { useAgent } from '@aries-framework/react-hooks'
import { agentDependencies } from '@aries-framework/react-native'
import React, { createContext, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { DeviceEventEmitter } from 'react-native'

import { EventTypes } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import {
  secretForPIN,
  storeWalletSecret,
  loadWalletSecret,
  loadWalletSalt,
  isBiometricsActive,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'
import { migrateToAskar, didMigrateToAskar } from '../utils/migration'

export interface AuthContext {
  checkPIN: (PIN: string) => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  removeSavedWalletSecret: () => void
  disableBiometrics: () => Promise<void>
  setPIN: (PIN: string) => Promise<void>
  commitPIN: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
  rekeyWallet: (oldPin: string, newPin: string, useBiometry?: boolean) => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const [store, dispatch] = useStore()
  const { agent } = useAgent()
  const { t } = useTranslation()

  const setPIN = async (PIN: string): Promise<void> => {
    const secret = await secretForPIN(PIN)
    await storeWalletSecret(secret)
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret && walletSecret.key) {
      return walletSecret
    }

    const { secret, err } = await loadWalletSecret(
      t('Biometry.UnlockPromptTitle'),
      t('Biometry.UnlockPromptDescription')
    )

    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, err !== undefined)

    if (!secret) {
      return
    }
    setWalletSecret(secret)

    return secret
  }

  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
    const secret = await getWalletCredentials()
    if (!secret) {
      return false
    }
    // set did authenticate to true if we can get wallet credentials
    dispatch({
      type: DispatchAction.DID_AUTHENTICATE,
    })
    if (useBiometry) {
      await storeWalletSecret(secret, useBiometry)
    } else {
      // erase wallet key if biometrics is disabled
      await wipeWalletKey(useBiometry)
    }
    return true
  }

  const checkPIN = async (PIN: string): Promise<boolean> => {
    try {
      const secret = await loadWalletSalt()

      if (!secret || !secret.salt) {
        return false
      }

      const hash = await hashPIN(PIN, secret.salt)

      if (!didMigrateToAskar(store.migration)) {
        await migrateToAskar(secret.id, hash)
        dispatch({
          type: DispatchAction.DID_MIGRATE_TO_ASKAR,
        })
      }

      // NOTE: a custom wallet is used to check if the wallet key is correct. This is different from the wallet used in the rest of the app.
      // We create an AskarWallet instance and open the wallet with the given secret.
      const askarWallet = new AskarWallet(
        new ConsoleLogger(LogLevel.off),
        new agentDependencies.FileSystem(),
        new SigningProviderRegistry([])
      )
      await askarWallet.open({
        id: secret.id,
        key: hash,
      })

      await askarWallet.close()

      const fullSecret = await secretForPIN(PIN, secret.salt)
      setWalletSecret(fullSecret)
      return true
    } catch (e) {
      return false
    }
  }

  const removeSavedWalletSecret = () => {
    setWalletSecret(undefined)
  }

  const disableBiometrics = async () => {
    await wipeWalletKey(true)
  }

  const rekeyWallet = async (oldPin: string, newPin: string, useBiometry?: boolean): Promise<boolean> => {
    try {
      // argon2.hash can sometimes generate an error
      const secret = await loadWalletSalt()
      if (!secret) {
        return false
      }
      const oldHash = await hashPIN(oldPin, secret.salt)
      const newSecret = await secretForPIN(newPin)
      const newHash = await hashPIN(newPin, newSecret.salt)
      if (!newSecret.key) {
        return false
      }

      await agent?.wallet.close()
      await agent?.wallet.rotateKey({ id: secret.id, key: oldHash, rekey: newHash })
      await storeWalletSecret(newSecret, useBiometry)
      setWalletSecret(newSecret)
    } catch {
      return false
    }
    return true
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        getWalletCredentials,
        removeSavedWalletSecret,
        disableBiometrics,
        commitPIN,
        setPIN,
        isBiometricsActive,
        rekeyWallet,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
