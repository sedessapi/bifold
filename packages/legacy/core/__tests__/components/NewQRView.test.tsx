import { useConnections } from '@aries-framework/react-hooks'
import { useAgent } from '@aries-framework/react-hooks'
import { act, render } from '@testing-library/react-native'
import React from 'react'

import NewQRView from '../../App/components/misc/NewQRView'
import { ConfigurationContext } from '../../App/contexts/configuration'
import { StoreProvider, defaultState } from '../../App/contexts/store'
import { testIdWithKey } from '../../App/utils/testable'
import { useNavigation } from '../../__mocks__/custom/@react-navigation/core'
import configurationContext from '../contexts/configuration'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

jest.mock('react-native-camera', () => {
  return require('../../__mocks__/custom/react-native-camera')
})

describe('NewQRView Component', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  beforeEach(() => {
    jest.clearAllMocks()
    // @ts-ignore
    useConnections.mockReturnValue({ records: [] })
  })

  const navigation = useNavigation()

  test('Renders correctly on first tab', async () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <NewQRView
          defaultToConnect={false}
          handleCodeScan={() => Promise.resolve()}
          navigation={navigation as any}
          route={{} as any}
        />
      </ConfigurationContext.Provider>
    )

    await act(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot()
  })

  test('Renders correctly on second tab', async () => {
    // @ts-ignore
    useAgent().agent?.oob.createInvitation.mockReturnValue({
      outOfBandInvitation: {
        toUrl: () => {
          return ''
        },
      },
    })
    const tree = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            walletName: 'My Wallet - 1234',
          },
        }}
      >
        <ConfigurationContext.Provider value={configurationContext}>
          <NewQRView
            defaultToConnect={true}
            handleCodeScan={() => Promise.resolve()}
            navigation={navigation as any}
            route={{} as any}
          />
        </ConfigurationContext.Provider>
      </StoreProvider>
    )

    await act(() => {
      jest.runAllTimers()
    })

    expect(tree).toMatchSnapshot()
  })

  test('Contains test IDs', async () => {
    // @ts-ignore
    useAgent().agent?.oob.createInvitation.mockReturnValue({
      outOfBandInvitation: {
        toUrl: () => {
          return ''
        },
      },
    })
    const { getByTestId } = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            walletName: 'My Wallet - 1234',
          },
        }}
      >
        <ConfigurationContext.Provider value={configurationContext}>
          <NewQRView
            defaultToConnect={true}
            handleCodeScan={() => Promise.resolve()}
            navigation={navigation as any}
            route={{} as any}
          />
        </ConfigurationContext.Provider>
      </StoreProvider>
    )

    await act(() => {
      jest.runAllTimers()
    })

    const editButtonByTestId = getByTestId(testIdWithKey('EditWalletName'))

    expect(editButtonByTestId).not.toBeNull()
  })
})
