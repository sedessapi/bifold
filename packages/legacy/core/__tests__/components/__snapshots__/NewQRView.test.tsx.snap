// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQRView Component Renders correctly on first tab 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#313132",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#313132",
        "borderTopWidth": 0,
        "flexDirection": "row",
        "height": 60,
        "paddingBottom": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      }
    }
  >
    <View
      accessibilityLabel="Scan.ScanQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.ScanQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="Scan.MyQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.MyQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`NewQRView Component Renders correctly on second tab 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#313132",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
            testID="com.ariesbifold:id/LoadingActivityIndicator"
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": "33%",
                  "width": "33%",
                }
              }
              testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
            />
            <View
              collapsable={false}
              style={
                Object {
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "360deg",
                    },
                  ],
                }
              }
            >
              <
                fill="#FFFFFF"
                height={200}
                width={200}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "marginBottom": 20,
                    "textAlign": "center",
                  },
                  Object {
                    "paddingHorizontal": 20,
                  },
                ]
              }
              testID="com.ariesbifold:id/WalletName"
            >
              My Wallet - 1234
            </Text>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
              testID="com.ariesbifold:id/EditWalletName"
            >
              <View
                accessibilityLabel="NameWallet.EditWalletName"
                accessible={true}
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 24,
                      },
                      Object {
                        "color": "#42803E",
                        "fontSize": 32,
                        "fontWeight": "bold",
                        "marginBottom": 20,
                        "marginLeft": 10,
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </RNGestureHandlerButton>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
                "textAlign": "center",
              }
            }
          >
            Connection.ShareQR
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#313132",
        "borderTopWidth": 0,
        "flexDirection": "row",
        "height": 60,
        "paddingBottom": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      }
    }
  >
    <View
      accessibilityLabel="Scan.ScanQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.ScanQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityLabel="Scan.MyQRCode"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Scan.MyQRCode"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#42803E",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
